bindings/java/fdbJNI.cpp

```diff
@@ -640,6 +640,150 @@ JNIEXPORT jlong JNICALL Java_com_apple_foundationdb_FDBTransaction_Transaction_1
 	return (jlong)f;
 }
 
+void getDirectRangeQueryResults(JNIEnv *jenv, FDBFuture* f, uint8_t* buffer, int bufferCapacity) {
+	const FDBKeyValue *kvs;
+	int count;
+	fdb_bool_t more;
+	fdb_error_t err = fdb_future_get_keyvalue_array( f, &kvs, &count, &more );
+	if( err ) {
+		safeThrow( jenv, getThrowable( jenv, err ) );
+		return;
+	}
+
+	// TODO (Vishesh): Should just fit whatever we can instead? FDB should ideally shouldn't give us
+	//    more bytes than we asked for.
```

Comment:
Byte limits are soft, so you can get more than you request. I think the way it works is that it will fetch kv-pairs up until the limit is exceeded and then stop, so you could exceed the limit by almost 1 whole kv-pair.